lib/heartcheck.rb

```diff
@@ -97,9 +102,11 @@ def info_checks
 
     # an executor class that respond to dispatch(checkers)
     #
-    # @return [Heartcheck::Executors::Base]
+    # @return [Heartcheck::Executors::Base] or [Heartcheck::Executors::HashResponse]
     def executor
-      @executor ||= Heartcheck::Executors::Base.new
```

Comment:
this boolean/default value strategy might cause a conflict/unexpected behaviour on the executor used if mixed with `use_threaded_executor!`, an user might not get the expected executor or have to choose between having the new out put format or the threaded executor.

IMHO, is feature breaks the single responsibility principle and formatting the result should be independent of the executor selected